### PR TITLE
Improve error message when property and callback declarations clash

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

**Comment by tronical** [comment]
I thought this was still allowed, but I tested on a wrong build :). This PR improves the error messages though (and adds what I think where missing "continue" statements).

File: internal/compiler/object_tree.rs [hunk 1]
```diff
@@ -741,10 +741,21 @@ impl Element {
                 ..
             } = r.lookup_property(&unresolved_prop_name);
             if !matches!(maybe_existing_prop_type, Type::Invalid) {
-                diag.push_error(
-                    format!("Cannot override property '{}'", prop_name),
-                    &prop_decl.DeclaredIdentifier().child_token(SyntaxKind::Identifier).unwrap(),
-                )
+                if matches!(maybe_existing_prop_type, Type::Callback { .. }) {
```
Comment (ogoffart) [inline internal/compiler/object_tree.rs:744]: You could siplify this by putting this before the parent if and do a else if.  or merge it with the parent if, in a match.

File: internal/compiler/object_tree.rs [hunk 2]
```diff
@@ -854,17 +865,25 @@ impl Element {
                 ..
             } = r.lookup_property(&name);
             if !matches!(maybe_existing_prop_type, Type::Invalid) {
-                if r.property_declarations.contains_key(&name) {
-                    diag.push_error(
-                        "Duplicated callback declaration".into(),
-                        &sig_decl.DeclaredIdentifier(),
-                    );
+                if matches!(maybe_existing_prop_type, Type::Callback { .. }) {
+                    if r.property_declarations.contains_key(&name) {
```
Comment (ogoffart) [inline internal/compiler/object_tree.rs:869]: I personally don't think we need to special case this,  but why not.

**Review by ogoffart — APPROVED** [review]
:+1:   
(comments are optionals)

File: internal/compiler/object_tree.rs (hunk 1 above)
Comment (tronical) [reply internal/compiler/object_tree.rs:744]: Good idea.

**Pull request merged**